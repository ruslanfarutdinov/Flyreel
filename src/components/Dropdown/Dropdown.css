.Dropdown {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.Select {
  border-radius: 4px;
}

.Label {
  margin: 5px 0;
}