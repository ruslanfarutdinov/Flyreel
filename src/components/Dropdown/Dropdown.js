import React from 'react';
import PropTypes from 'prop-types';

import './Dropdown.css';

function Dropdown({ 
  selectId,
  labelText,
  options,
  onChangeHandler,
 }) {
  return (
    <div className="Dropdown">
      <label htmlFor={selectId}>{labelText}</label>
      <select id={selectId} onChange={onChangeHandler}>
        <option value="">-Please choose an option-</option>
        {options.map(option => (
          <option key={option.id} value={option.value}>{option.value}</option>
        ))}
      </select>
    </div>
  );
}

Dropdown.propTypes = {
  selectId: PropTypes.string.isRequired,
  labelText: PropTypes.string.isRequired,
  options: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      value: PropTypes.string,
    })).isRequired,
  onChangeHandler: PropTypes.func.isRequired,
};

export default Dropdown;