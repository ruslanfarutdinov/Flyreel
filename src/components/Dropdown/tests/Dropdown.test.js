import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import Dropdown from '../Dropdown';
import { selectId, labelText, options } from './mockData';

const mockOnOptionChange = jest.fn();

const renderDropdown = () =>
  render(
    <Dropdown
      selectId={selectId}
      labelText={labelText}
      options={options}
      onOptionChange={mockOnOptionChange}
    />
  );

describe('<Dropdown />', () => {
  it('should match snapshot', () => {
    const { asFragment } = renderDropdown();

    expect(asFragment()).toMatchSnapshot();
  });

  it('should have correct select element id', () => {
    renderDropdown();

    expect(document.getElementById(selectId)).toBeInTheDocument();
  });

  it('should display correct label text', () => {
    const { getByTestId } = renderDropdown();

    expect(getByTestId('label').textContent).toBe(labelText);
  });

  it('should display correct selected option', () => {
    renderDropdown();
    const selectElement = document.getElementById(selectId);

    expect(selectElement.value).toBe(options[0].value);

    fireEvent.change(
      selectElement, 
      { target: { value: 'Option 3' } },
    );
    expect(selectElement.value).toBe(options[2].value);
  });

  it('should fire onChange when an option changes', () => {
    renderDropdown();
    const selectElement = document.getElementById(selectId);

    fireEvent.change(
      selectElement, 
      { target: { value: 'Option 3' } },
    );

    expect(mockOnOptionChange).toHaveBeenCalledTimes(1);
  });
});
