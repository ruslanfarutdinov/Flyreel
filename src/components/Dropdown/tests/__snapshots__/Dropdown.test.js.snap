// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="Dropdown"
  >
    <label
      class="Label"
      data-testid="label"
      for="selectId"
    >
      labelText
    </label>
    <select
      class="Select"
      id="selectId"
    >
      <option
        value="Option 1"
      >
        Option 1
      </option>
      <option
        value="Option 2"
      >
        Option 2
      </option>
      <option
        value="Option 3"
      >
        Option 3
      </option>
    </select>
  </div>
</DocumentFragment>
`;
